test/spec/test-common.js

```diff
@@ -837,6 +837,130 @@ describe("tus", function () {
       expect(options.onSuccess).toHaveBeenCalled();
     });
 
+    // This tests ensures that tus-js-client correctly retries if the
+    // response of the onShouldRetry is true
+    it("should retry the upload when callback specified and returns true", async function () {
+      const testStack = new TestHttpStack();
+      var file = getBlob("hello world");
+      var options = {
+        httpStack: testStack,
+        endpoint: "http://tus.io/files/",
+        retryDelays: [10, 10, 10],
+        onSuccess: waitableFunction("onSuccess"),
+        onShouldRetry: () => true
```

Comment:
Can you add assertions that onShouldRetry is called with the correct parameters? You can do that using the spyOn functions.